.github/workflows/ci.yaml

```
@@ -0,0 +1,53 @@
+name: CI Testing
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  build:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: ["3.6", "3.7", "3.8", "3.9"]
+        os:  [ubuntu-latest]
+        # os: [ubuntu-latest, windows-latest, macos-latest]
+        include:
+          - os: windows-latest
+            python-version: "3.9"
+          - os: macos-latest
+            python-version: "3.9"
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade setuptools pip wheel black codecov
+        python -m pip install .[dev]
+    - name: Lint with black
+      run: |
+        black --check .
+    - name: Test with pytest
+      run: |
+
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v1
+      env:
+        OS: ${{ matrix.os }}
+        PYTHON: ${{ matrix.python-version }}
+      with:
+        # token: ${{ secrets.CODECOV_TOKEN }}  # not required for public repos
+        env_vars: OS,PYTHON
+        fail_ci_if_error: true
+#    - name: Notifications
+#      uses: juztcode/gitter-github-action@v1
+#      if: ${{ failure() }}
+#      with:
+#        room-id: ${{ secrets.GITTER_ROOM_ID }}
+#        token: ${{ secrets.GITTER_TOKEN }}
+#        text: sentinelsat build failure
```

Gitter secrets are in place, so you can activate this part.

```suggestion
      uses: juztcode/gitter-github-action@v1
      if: ${{ failure() }}
      with:
        room-id: ${{ secrets.GITTER_ROOM_ID }}
        token: ${{ secrets.GITTER_TOKEN }}
        text: sentinelsat build failure
```

---

This is done now but it seems not to work
```
Error: Request failed with status code 400
```